src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py

```diff
@@ -526,6 +527,28 @@ def test_functionapp_windows_runtime(self, resource_group, storage_account):
         self.cmd('functionapp config appsettings list -g {} -n {}'.format(resource_group, functionapp_name), checks=[
             JMESPathCheck("[?name=='FUNCTIONS_WORKER_RUNTIME'].value|[0]", 'node')])
 
+    @ResourceGroupPreparer(location=FUNCTIONAPP_EUAP_LOCATION)
+    @StorageAccountPreparer()
+    def test_functionapp_create_with_appcontainer_managed_environment(self, resource_group, storage_account):
+        functionapp_name = self.create_random_name(
+            'functionappwindowsruntime', 40)
+        managed_environment_name = self.create_random_name(
+            'containerappmanagedenvironment', 40
+        )
+
+        self.cmd('containerapp env create --name {} --resource-group {} --location {}'
+        .format(managed_environment_name, resource_group, FUNCTIONAPP_EUAP_LOCATION)).assert_with_checks([
+                     JMESPathCheck('name', managed_environment_name),
+                     JMESPathCheck('resourceGroup', resource_group),
+                     JMESPathCheck('location', FUNCTIONAPP_EUAP_LOCATION)])
+
+        self.cmd('functionapp create -g {} -n {} -c {} -s {} --environment {} --os-type Windows --functions-version 4'
+                 .format(resource_group, functionapp_name, WINDOWS_ASP_LOCATION_FUNCTIONAPP, storage_account, managed_environment_name)).assert_with_checks([
+                     JMESPathCheck('state', 'Running'),
+                     JMESPathCheck('name', functionapp_name),
+                     JMESPathCheck('kind', 'functionapp'),
+                     JMESPathCheck('hostNames[0]', functionapp_name + '.azurewebsites.net')])
+
```

Comment:
When the backend changes are released, let's also add as part of this unit test the `functionapp show -g {} -n {}` command to check that the managed environment parameter is passed back.

Reply:
This is a TODO for later